File: comment_service/python/controllers/commentController.py
from flask import jsonify, request
from pymongo import MongoClient
from bson.objectid import ObjectId
from datetime import datetime

client = MongoClient('mongodb://localhost:27017/')
db = client['social_media_platform']
comments = db['comments']

def get_comments():
    all_comments = list(comments.find())
    return jsonify(all_comments)

def create_comment(req):
    data = req.get_json()
    comment = {
        "postId": ObjectId(data['postId']),
        "userId": ObjectId(data['userId']),
        "content": data['content'],
        "createdAt": datetime.utcnow()
    }
    comments.insert_one(comment)
    return jsonify(comment), 201

File: services/comment_service/controllers/commentController.js
const { MongoClient, ObjectId } = require('mongodb');
const url = 'mongodb://localhost:27017';
const dbName = 'social_media_platform';

const getComments = async (req, res) => {
  const client = new MongoClient(url);
  try {
    await client.connect();
    const db = client.db(dbName);
    const comments = await db.collection('comments').find().toArray();
    res.json(comments);
  } finally {
    client.close();
  }
};

const createComment = async (req, res) => {
  const client = new MongoClient(url);
  const { postId, userId, content } = req.body;
  try {
    await client.connect();
    const db = client.db(dbName);
    const result = await db.collection('comments').insertOne({
      postId: ObjectId(postId),
      userId: ObjectId(userId),
      content,
      createdAt: new Date()
    });
    res.status(201).json(result.ops[0]);
  } finally {
    client.close();
  }
};

module.exports = {
  getComments,
  createComment
};
